File: Strategy Summit/Number Theory/Q2. The kth Factor of n/Q2_ThekthFactorofn.java
class Solution {
    public int kthFactor(int n, int k) {
        // Counter to track how many factors of n we have found so far
        int count = 0;

        // Loop from 1 to n (inclusive) to check all possible factors
        for (int i = 1; i <= n; i++) {
            
            // Check if i divides n completely (i is a factor of n)
            if (n % i == 0) {
                
                // Increment the count of factors
                count++;
                
                // If this is the k-th factor, return it immediately
                if (count == k)
                    return i;
            }
        }

        // If n has fewer than k factors, return -1
        return -1;
    }
}

File: Strategy Summit/Number Theory/Q2. The kth Factor of n/Q2_ThekthFactorofn.py
class Solution(object):
    def kthFactor(self, n, k):
        # Initialize a counter to keep track of factors found
        count = 0
        
        # Loop from 1 to n (inclusive) to check all possible factors
        for i in range(1, n + 1):
            
            # Check if i is a factor of n
            if n % i == 0:
                # Increment the factor count
                count += 1
                
                # If this is the k-th factor, return it
                if count == k:
                    return i
        
        # If fewer than k factors exist, return -1
        return -1